src/wasm/wat-parser.cpp

```diff
@@ -250,6 +269,17 @@ struct ParseInput {
     auto curr = getPos();
     return std::string_view(prev.pos, curr.pos - prev.pos);
   }
+
+  [[nodiscard]] Err err(std::string reason) {
+    std::stringstream msg;
+    if (auto t = peek()) {
+      msg << lexer.position(*t);
+    } else {
+      msg << lexer.position(lexer.next());
+    }
```

Comment:
I'm having trouble understanding what this if-else does. `peek` returns a value if there is a value left, so in that case we return the position of that value, but what is the else doing? What is "next" if there is nothing to peek at?

Overall the general shape of error handling looks very nice!

Reply:
I see, thanks!